onnxscript/rewriter/pattern.py

```diff
@@ -1448,8 +1452,8 @@ def rewrite(cls, op, *_) -> Any:
         raise NotImplementedError("Method 'rewrite' must be overwritten.")
 
     @classmethod
-    def check(cls, context, *_, **__) -> bool:
-        return True
+    def check(cls, context, *_, **__) -> MatchResult:
```

Comment:
nit: I think allowing check to return `bool | MatchResult` would help with some backward compatibility (especially if there are external users using this). There is nothing wrong with `True` in successful cases. The calling logic within the pattern matcher can handle all the conversion.